File: service/internal/users/sql/repository.go
package sql

import "github.com/sazzer/nrl/service/internal/infrastructure/database"

// The repository of user data.
type repository struct {
	db database.Database
}

// Create a new users repository.
// nolint:golint
// This is deliberately unexported so it can only be used by the service layer.
func New(db database.Database) repository {
	return repository{db}
}

File: service/src/users/repository.rs
mod create;
mod errors;
mod get;
mod model;

use crate::infrastructure::database::Database;
pub use errors::*;
use std::sync::Arc;

/// Repository of users data.
pub struct UsersRepository {
    database: Arc<Database>,
}

impl UsersRepository {
    /// Create a new users repository.
    ///
    /// # Parameters
    /// - `database` - The database connection
    pub fn new(database: Arc<Database>) -> Self {
        Self { database }
    }
}
